backend/controllers/appointments/viewAppointmentsController.js

```diff
@@ -0,0 +1,98 @@
+const moment = require('moment')
+const {
+  app: { notmEndpointUrl: dpsUrl },
+} = require('../../config')
+const { serviceUnavailableMessage } = require('../../common-messages')
+const { getTime, properCaseName, formatName } = require('../../utils')
+
+module.exports = ({ elite2Api, whereaboutsApi, logError }) => async (req, res) => {
+  const { date, timeSlot = 'AM', type, locationId } = req.query
+  const searchDate = date ? moment(date, 'DD/MM/YYYY').format('YYYY-MM-DD') : moment().format('YYYY-MM-DD')
+  const agencyId = req.session.userDetails.activeCaseLoadId
+
+  try {
+    const [appointmentTypes, appointmentLocations, appointments] = await Promise.all([
+      elite2Api.getAppointmentTypes(res.locals),
+      elite2Api.getLocationsForAppointments(res.locals, agencyId),
+      elite2Api.getAppointmentsForAgency(res.locals, {
+        agencyId,
+        date: searchDate,
+        timeSlot: timeSlot !== 'all' ? timeSlot : undefined,
+        locationId,
+      }),
+    ])
+
+    const videoLinkAppointmentIds = appointments
+      .filter(appointment => appointment.appointmentTypeCode === 'VLB')
+      .map(videoLinkAppointment => videoLinkAppointment.id)
+
+    const videoLinkAppointments = await whereaboutsApi.getVideoLinkAppointments(res.locals, videoLinkAppointmentIds)
+
+    const appointmentsEnhanced = appointments
+      .filter(appointment => (type ? appointment.appointmentTypeCode === type : true))
+      .map(async appointment => {
+        const { startTime, endTime, offenderNo } = appointment
+        const staffDetails = await elite2Api.getStaffDetails(res.locals, appointment.auditUserId).catch(error => {
```

Comment:
Slightly concerned about performance here because it would make loads of requests for staff details. We'll have to keep an eye on it.